library/src/blas1/rocblas_copy_batched.cpp

```diff
@@ -0,0 +1,149 @@
+/* ************************************************************************
+ * Copyright 2016-2019 Advanced Micro Devices, Inc.
+ * ************************************************************************ */
+#include "handle.h"
+#include "logging.h"
+#include "rocblas.h"
+#include "rocblas_copy.hpp"
+#include "utility.h"
+
+namespace
+{
+    template <typename>
+    constexpr char rocblas_copy_batched_name[] = "unknown";
+    template <>
+    constexpr char rocblas_copy_batched_name<float>[] = "rocblas_scopy_batched";
+    template <>
+    constexpr char rocblas_copy_batched_name<double>[] = "rocblas_dcopy_batched";
+    template <>
+    constexpr char rocblas_copy_batched_name<rocblas_half>[] = "rocblas_hcopy_batched";
+    template <>
+    constexpr char rocblas_copy_batched_name<rocblas_float_complex>[] = "rocblas_ccopy_batched";
+    template <>
+    constexpr char rocblas_copy_batched_name<rocblas_double_complex>[] = "rocblas_zcopy_batched";
+
+    template <rocblas_int NB, typename T>
+    rocblas_status rocblas_copy_batched_impl(rocblas_handle handle,
+                                             rocblas_int    n,
+                                             const T* const x[],
+                                             rocblas_int    incx,
+                                             T* const       y[],
+                                             rocblas_int    incy,
+                                             rocblas_int    batch_count)
+    {
+        if(!handle)
+            return rocblas_status_invalid_handle;
+
+        auto layer_mode = handle->layer_mode;
+        if(layer_mode & rocblas_layer_mode_log_trace)
+            log_trace(handle, rocblas_copy_batched_name<T>, n, x, incx, y, incy, batch_count);
+
+        if(layer_mode & rocblas_layer_mode_log_bench)
+            log_bench(handle,
+                      "./rocblas-bench -f copy_batched -r",
+                      rocblas_precision_string<T>,
+                      "-n",
+                      n,
+                      "--incx",
+                      incx,
+                      "--incy",
+                      incy,
+                      "batch_count",
+                      batch_count);
+
+        if(layer_mode & rocblas_layer_mode_log_profile)
+            log_profile(handle,
+                        rocblas_copy_batched_name<T>,
+                        "N",
+                        n,
+                        "incx",
+                        incx,
+                        "incy",
+                        incy,
+                        "batch_count",
+                        batch_count);
+
+        if(!x || !y)
+            return rocblas_status_invalid_pointer;
+
+        RETURN_ZERO_DEVICE_MEMORY_SIZE_IF_QUERIED(handle);
+
+        if(!incx || !incy || batch_count < 0)
+            return rocblas_status_invalid_size;
```

Comment:
`incx==0 || incy ==0` should be okay. http://www.netlib.org/lapack/lapack-3.1.1/html/dcopy.f.html